src/Widgets/ActionBar.vala

```diff
@@ -0,0 +1,52 @@
+public class Spreadsheet.Widgets.ActionBar : Gtk.ActionBar {
+    public signal void zoom_level_changed ();
+
+    private Gtk.Adjustment zoom_scale_adj;
+
+    public int zoom_level {
+        get {
+            return App.settings.get_int ("zoom-level");
+        }
+        set {
+            zoom_scale_adj.value = value;
+            App.settings.set_int ("zoom-level", value);
+        }
+    }
+
+    public string zoom_level_text {
+        owned get {
+            return "%i %%".printf (zoom_level);
+        }
+    }
+
+    public ActionBar () {
```

Comment:
Is this constructor really needed? Doesn't the compiler adds it for you?

Reply:
You're right. I added this but it's not really necessary.